honeybadger-java/src/main/java/io/honeybadger/reporter/config/ConfigContext.java

```diff
@@ -12,7 +12,7 @@
  */
 public interface ConfigContext {
     /** Global reference to the currently used config per thread. */
-    ThreadLocal<ConfigContext> threadLocal = new ThreadLocal<>();
+    ThreadLocal<ConfigContext> THREAD_LOCAL = new ThreadLocal<>();
```

Comment:
After your refactors, I think this field is no longer needed and can be safely deleted. For a good reason too! The original implementation was wacky and was there for expediency.